### PR TITLE
Dotnet identification with one token+ registration

### PR DESCRIPTION
**Pull request by Melo5154**

### CODE REVIEW

File: GdscBackend/Authentication/RegisterViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+using System.ComponentModel.DataAnnotations; 
+
+namespace gdsc_web_backend.Authentication
+{
+    public class RegisterViewModel
+    {
+        [Required(ErrorMessage = "User Name is required")]
```
Comment (alexgaita) [inline GdscBackend/Authentication/RegisterViewModel.cs:7]: Bad indentation

File: GdscBackend/Authentication/LoginViewModel.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+{
+    public class LoginViewModel
+    {
+        [Required(ErrorMessage = "User Name is required")]  
+        public string Username { get; set; }  
+  
+        [Required(ErrorMessage = "Password is required")]  
+        public string Password { get; set; } 
+    }
+}
```
Comment (alexgaita) [inline GdscBackend/Authentication/LoginViewModel.cs:13]: Add newline to end of file, search in all files for this mistake.

File: GdscBackend/Authentication/Role.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+using System;
+using Microsoft.AspNetCore.Identity;
+
+namespace gdsc_web_backend.Authentication
+{
+    public class Role:IdentityRole<string>
```
Comment (alexgaita) [inline GdscBackend/Authentication/Role.cs:6]: add spaces around colon.

File: GdscBackend/Authentication/Role.cs [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    {
+
```
Comment (alexgaita) [inline GdscBackend/Authentication/Role.cs:8]: newline not needed

File: GdscBackend/Controllers/v1/AuthenticationController.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+    [Route("api/v1/auth")]
+    [ApiController]
+    public class AuthenticationController : ControllerBase
+    {
+        private readonly UserManager<User> userModel;
+        private readonly IConfiguration _configuration;
+
+        public AuthenticationController(UserManager<User> userModel, IConfiguration configuration)
+        {
+            this.userModel = userModel;
```
Comment (alexgaita) [inline GdscBackend/Controllers/v1/AuthenticationController.cs:24]: "this" not needed

File: GdscBackend/Controllers/v1/AuthenticationController.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        public async Task<IActionResult> Login([FromBody] LoginViewModel model)
+        {
+            var user = await userModel.FindByNameAsync(model.Username);
+            if (user != null && await userModel.CheckPasswordAsync(user, model.Password))
+            {
+                var userRoles = await userModel.GetRolesAsync(user);
+
+                var authClaims = new List<Claim>
+                {
+                    new Claim(ClaimTypes.Name, user.UserName),
```
Comment (alexgaita) [inline GdscBackend/Controllers/v1/AuthenticationController.cs:39]: Claim not needed because you already have a list of Claim Objects

Comment (alexgaita) [reply GdscBackend/Controllers/v1/AuthenticationController.cs:39]: Use new ( ClaimTypes.Name, .... )

**Review by alexgaita — CHANGES_REQUESTED** [review]

File: GdscBackend/Controllers/v1/AuthenticationController.cs (hunk 5 above)
Comment (Melo5154) [reply GdscBackend/Controllers/v1/AuthenticationController.cs:24]: este nevoie de acel this pentru ca programul nu va stii la care userModel atribuie acea valoare de userModel

**Pull request merged**